File: app/src/main/java/wxm/KeepAccount/define/UsrItem.java
package wxm.KeepAccount.define;

import com.j256.ormlite.field.DataType;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import wxm.androidutil.DBHelper.IDBRow;

/**
 * usr class
 * Created by WangXM on 2016/8/5.
 */
@DatabaseTable(tableName = "tbUsr")
public class UsrItem
        implements IDBRow<Integer> {
    public final static String FIELD_ID = "_id";
    public final static String FIELD_NAME = "name";
    public final static String FIELD_PWD = "pwd";

    @DatabaseField(generatedId = true, columnName = "_id", dataType = DataType.INTEGER)
    private int _id;

    @DatabaseField(columnName = "name", unique = true, dataType = DataType.STRING)
    private String name;

    @DatabaseField(columnName = "pwd", dataType = DataType.STRING)
    private String pwd;

    public UsrItem() {
        setName("");
        setPwd("");
        setId(-1);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPwd() {
        return pwd;
    }

    public void setPwd(String pwd) {
        this.pwd = pwd;
    }

    public int getId() {
        return _id;
    }

    public void setId(int _id) {
        this._id = _id;
    }

    @Override
    public Integer getID() {
        return getId();
    }

    @Override
    public void setID(Integer integer) {
        setId(integer);
    }
}

File: app/src/main/java/wxm/KeepAccount/define/UsrItem.kt
package wxm.KeepAccount.define

import com.j256.ormlite.field.DataType
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import wxm.androidutil.db.IDBRow


/**
 * usr class
 * Created by WangXM on 2016/8/5.
 */
@DatabaseTable(tableName = "tbUsr")
class UsrItem : IDBRow<Int>, Cloneable, IPublicClone {

    @DatabaseField(generatedId = true, columnName = FIELD_ID, dataType = DataType.INTEGER)
    var id: Int = GlobalDef.INVALID_ID

    @DatabaseField(columnName = FIELD_NAME, unique = true, dataType = DataType.STRING)
    var name: String? = ""

    @DatabaseField(columnName = FIELD_PWD, dataType = DataType.STRING)
    var pwd: String? = ""

    override fun getID(): Int {
        return id
    }

    override fun setID(integer: Int) {
        id = integer
    }

    override fun clone(): Any {
        return UsrItem().let{
            it.id = this.id
            it.name = this.name
            it.pwd = this.pwd

            it
        }
    }

    override fun publicClone(): Any {
        return clone()
    }

    companion object {
        const val FIELD_ID = "_id"
        const val FIELD_NAME = "name"
        const val FIELD_PWD = "pwd"
    }
}
